.github/workflows/bandit_security_test.yml

```
@@ -0,0 +1,27 @@
+name: Security check - Bandit  # Name of the GitHub Actions workflow
+
+on: [push, pull_request]  # Trigger the workflow 
+
+jobs:
+  build:
+    runs-on: ubuntu-latest  # Executes the job on the latest version of Ubuntu
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]  # Running matrix jobs on both Ubuntu and macOS
+    name: Python ${{ matrix.os }}  # Name the job based on the OS being used
+
+    steps:
+    - uses: actions/checkout@v2  # Checks out your repository's code
+
+    - name: Security check - Bandit  # Run Bandit security check
+      uses: ioggstream/bandit-report-artifacts@v0.0.2  # Using Bandit for security checks
+      with:
+        project_path: .  # Path to the project to scan
+        ignore_failure: true  # Continue the workflow even if Bandit reports issues
```

we obviously dont want to ignore errors and failures. we want the checks to fail on encountering such errors